Warnings

~~~
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc: In instantiation of ‘void deal2lkit::{anonymous}::default_create_grid(dealii::Triangulation<dim, spacedim>&, std::string, dealii::Point<spacedim>, dealii::Point<spacedim>, double, double, double, bool, unsigned int, unsigned int, std::vector<unsigned int>, std::string) [with int dim = 3; int spacedim = 3; std::string = std::basic_string<char>]’:
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:538:60:   required from ‘void deal2lkit::{anonymous}::create_grid(dealii::Triangulation<dim, dim>&, std::string, dealii::Point<dim>, dealii::Point<dim>, double, double, double, bool, unsigned int, unsigned int, std::vector<unsigned int>, std::string) [with int dim = 3; std::string = std::basic_string<char>]’
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:580:45:   required from here
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_two’ [-Wunused-parameter]
   default_create_grid( Triangulation<dim,spacedim> &tria,
   ^
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_one’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_three’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘un_int_option_one’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘un_int_option_two’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc: In instantiation of ‘void deal2lkit::{anonymous}::default_create_grid(dealii::Triangulation<dim, spacedim>&, std::string, dealii::Point<spacedim>, dealii::Point<spacedim>, double, double, double, bool, unsigned int, unsigned int, std::vector<unsigned int>, std::string) [with int dim = 2; int spacedim = 3; std::string = std::basic_string<char>]’:
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:413:62:   required from ‘void deal2lkit::{anonymous}::create_grid(dealii::Triangulation<dim, (dim + 1)>&, std::string, dealii::Point<(dim + 1)>, dealii::Point<(dim + 1)>, double, double, double, bool, unsigned int, unsigned int, std::vector<unsigned int>, std::string) [with int dim = 2; std::string = std::basic_string<char>]’
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:619:43:   required from here
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_two’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_one’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_three’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘un_int_option_one’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘un_int_option_two’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc: In instantiation of ‘void deal2lkit::{anonymous}::default_create_grid(dealii::Triangulation<dim, spacedim>&, std::string, dealii::Point<spacedim>, dealii::Point<spacedim>, double, double, double, bool, unsigned int, unsigned int, std::vector<unsigned int>, std::string) [with int dim = 1; int spacedim = 1; std::string = std::basic_string<char>]’:
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:538:60:   required from ‘void deal2lkit::{anonymous}::create_grid(dealii::Triangulation<dim, dim>&, std::string, dealii::Point<dim>, dealii::Point<dim>, double, double, double, bool, unsigned int, unsigned int, std::vector<unsigned int>, std::string) [with int dim = 1; std::string = std::basic_string<char>]’
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:645:36:   required from ‘void deal2lkit::ParsedGridGenerator<dim, spacedim>::create(dealii::Triangulation<dim, spacedim>&) [with int dim = 1; int spacedim = 1]’
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:732:27:   required from here
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_two’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_one’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_three’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘un_int_option_one’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘un_int_option_two’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc: In instantiation of ‘void deal2lkit::{anonymous}::default_create_grid(dealii::Triangulation<dim, spacedim>&, std::string, dealii::Point<spacedim>, dealii::Point<spacedim>, double, double, double, bool, unsigned int, unsigned int, std::vector<unsigned int>, std::string) [with int dim = 1; int spacedim = 2; std::string = std::basic_string<char>]’:
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:413:62:   required from ‘void deal2lkit::{anonymous}::create_grid(dealii::Triangulation<dim, (dim + 1)>&, std::string, dealii::Point<(dim + 1)>, dealii::Point<(dim + 1)>, double, double, double, bool, unsigned int, unsigned int, std::vector<unsigned int>, std::string) [with int dim = 1; std::string = std::basic_string<char>]’
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:645:36:   required from ‘void deal2lkit::ParsedGridGenerator<dim, spacedim>::create(dealii::Triangulation<dim, spacedim>&) [with int dim = 1; int spacedim = 2]’
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:733:27:   required from here
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_two’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_one’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_three’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘un_int_option_one’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘un_int_option_two’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc: In instantiation of ‘void deal2lkit::{anonymous}::default_create_grid(dealii::Triangulation<dim, spacedim>&, std::string, dealii::Point<spacedim>, dealii::Point<spacedim>, double, double, double, bool, unsigned int, unsigned int, std::vector<unsigned int>, std::string) [with int dim = 2; int spacedim = 2; std::string = std::basic_string<char>]’:
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:538:60:   required from ‘void deal2lkit::{anonymous}::create_grid(dealii::Triangulation<dim, dim>&, std::string, dealii::Point<dim>, dealii::Point<dim>, double, double, double, bool, unsigned int, unsigned int, std::vector<unsigned int>, std::string) [with int dim = 2; std::string = std::basic_string<char>]’
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:645:36:   required from ‘void deal2lkit::ParsedGridGenerator<dim, spacedim>::create(dealii::Triangulation<dim, spacedim>&) [with int dim = 2; int spacedim = 2]’
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:734:27:   required from here
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_two’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_one’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_three’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘un_int_option_one’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘un_int_option_two’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc: In instantiation of ‘void deal2lkit::{anonymous}::default_create_grid(dealii::Triangulation<dim, spacedim>&, std::string, dealii::Point<spacedim>, dealii::Point<spacedim>, double, double, double, bool, unsigned int, unsigned int, std::vector<unsigned int>, std::string) [with int dim = 3; int spacedim = 3; std::string = std::basic_string<char>]’:
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:538:60:   required from ‘void deal2lkit::{anonymous}::create_grid(dealii::Triangulation<dim, dim>&, std::string, dealii::Point<dim>, dealii::Point<dim>, double, double, double, bool, unsigned int, unsigned int, std::vector<unsigned int>, std::string) [with int dim = 3; std::string = std::basic_string<char>]’
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:580:45:   required from here
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_two’ [-Wunused-parameter]
   default_create_grid( Triangulation<dim,spacedim> &tria,
   ^
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_one’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_three’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘un_int_option_one’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘un_int_option_two’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc: In instantiation of ‘void deal2lkit::{anonymous}::default_create_grid(dealii::Triangulation<dim, spacedim>&, std::string, dealii::Point<spacedim>, dealii::Point<spacedim>, double, double, double, bool, unsigned int, unsigned int, std::vector<unsigned int>, std::string) [with int dim = 2; int spacedim = 3; std::string = std::basic_string<char>]’:
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:413:62:   required from ‘void deal2lkit::{anonymous}::create_grid(dealii::Triangulation<dim, (dim + 1)>&, std::string, dealii::Point<(dim + 1)>, dealii::Point<(dim + 1)>, double, double, double, bool, unsigned int, unsigned int, std::vector<unsigned int>, std::string) [with int dim = 2; std::string = std::basic_string<char>]’
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:619:43:   required from here
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_two’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_one’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_three’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘un_int_option_one’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘un_int_option_two’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc: In instantiation of ‘void deal2lkit::{anonymous}::default_create_grid(dealii::Triangulation<dim, spacedim>&, std::string, dealii::Point<spacedim>, dealii::Point<spacedim>, double, double, double, bool, unsigned int, unsigned int, std::vector<unsigned int>, std::string) [with int dim = 1; int spacedim = 1; std::string = std::basic_string<char>]’:
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:538:60:   required from ‘void deal2lkit::{anonymous}::create_grid(dealii::Triangulation<dim, dim>&, std::string, dealii::Point<dim>, dealii::Point<dim>, double, double, double, bool, unsigned int, unsigned int, std::vector<unsigned int>, std::string) [with int dim = 1; std::string = std::basic_string<char>]’
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:645:36:   required from ‘void deal2lkit::ParsedGridGenerator<dim, spacedim>::create(dealii::Triangulation<dim, spacedim>&) [with int dim = 1; int spacedim = 1]’
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:732:27:   required from here
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_two’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_one’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_three’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘un_int_option_one’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘un_int_option_two’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc: In instantiation of ‘void deal2lkit::{anonymous}::default_create_grid(dealii::Triangulation<dim, spacedim>&, std::string, dealii::Point<spacedim>, dealii::Point<spacedim>, double, double, double, bool, unsigned int, unsigned int, std::vector<unsigned int>, std::string) [with int dim = 1; int spacedim = 2; std::string = std::basic_string<char>]’:
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:413:62:   required from ‘void deal2lkit::{anonymous}::create_grid(dealii::Triangulation<dim, (dim + 1)>&, std::string, dealii::Point<(dim + 1)>, dealii::Point<(dim + 1)>, double, double, double, bool, unsigned int, unsigned int, std::vector<unsigned int>, std::string) [with int dim = 1; std::string = std::basic_string<char>]’
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:645:36:   required from ‘void deal2lkit::ParsedGridGenerator<dim, spacedim>::create(dealii::Triangulation<dim, spacedim>&) [with int dim = 1; int spacedim = 2]’
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:733:27:   required from here
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_two’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_one’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_three’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘un_int_option_one’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘un_int_option_two’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc: In instantiation of ‘void deal2lkit::{anonymous}::default_create_grid(dealii::Triangulation<dim, spacedim>&, std::string, dealii::Point<spacedim>, dealii::Point<spacedim>, double, double, double, bool, unsigned int, unsigned int, std::vector<unsigned int>, std::string) [with int dim = 2; int spacedim = 2; std::string = std::basic_string<char>]’:
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:538:60:   required from ‘void deal2lkit::{anonymous}::create_grid(dealii::Triangulation<dim, dim>&, std::string, dealii::Point<dim>, dealii::Point<dim>, double, double, double, bool, unsigned int, unsigned int, std::vector<unsigned int>, std::string) [with int dim = 2; std::string = std::basic_string<char>]’
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:645:36:   required from ‘void deal2lkit::ParsedGridGenerator<dim, spacedim>::create(dealii::Triangulation<dim, spacedim>&) [with int dim = 2; int spacedim = 2]’
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:734:27:   required from here
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_two’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_one’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘double_option_three’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘un_int_option_one’ [-Wunused-parameter]
/home/heltai/src/deal2lkit/source/parsed_grid_generator.cc:333:3: warning: unused parameter ‘un_int_option_two’ [-Wunused-parameter]
~~~

--- a/include/deal2lkit/fe_values_cache.h
+++ b/include/deal2lkit/fe_values_cache.h
@@ -116,7 +116,7 @@ public:
            ExcMessage("You did not call cache_local_solution_vector with the right types!"));
 
     return  cache.template get<std::vector<Number> >(dofs_name);
-  };
+  }
 
 
   /**

--- a/source/parsed_grid_generator.cc
+++ b/source/parsed_grid_generator.cc
@@ -371,6 +371,12 @@ namespace
         else
           Assert(false, ExcNotImplemented());
       }
+    // silence warnings of unused variables
+    (void)double_option_one;
+    (void)double_option_two;
+    (void)double_option_three;
+    (void)un_int_option_one;
+    (void)un_int_option_two;
   }
 
   /**

